Upgraded test setup to support the latest ESLint version for new npm release.

MarkusBordihn opened the pull request:
Upgraded test setup to support the latest ESLint version. #75
Added automated tests for configuration validation.  #73
Prepared package for npm release. #76
Bumped major version to reflect breaking changes.
Fixed formatting issues in `README.md`.

MarkusBordihn committed:
Fixed formatting issues in `README.md`.
Upgraded test setup to support the latest ESLint version.
Added automated tests for configuration validation.
Prepared package for npm release.
Bumped major version to reflect breaking changes.

iennae commented on .github/workflows/unit_tests.yml:
```diff
@@ -0,0 +1,33 @@
+
+    steps:
+      - name: Check out Git repository
+        uses: actions/checkout@v3
```
let's pin to a commit hash for recommended security practices 

actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 (this is v4) as an example 

https://github.com/GoogleCloudPlatform/nodejs-docs-samples/blob/main/.github/workflows/custard-ci.yaml#L44C15-L44C72

iennae commented on .github/workflows/unit_tests.yml:
```diff
@@ -0,0 +1,33 @@
+          submodules: true
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
```
pin hash here 

uses: actions/setup-node@49933ea5288caeca8642d1e84afbd3f7d6820020 # v4

iennae reviewed (approved):
other than pinning via hash I think this is good to go. Thank you!

MarkusBordihn committed:
Update ESLint and related dependencies to latest versions.
Pin to a commit hash for automated workflows for recommended security practices.

MarkusBordihn commented:
Thank you for the feedback! I've addressed all comments, and the workflow tests on my forked repository completed successfully without any errors:
https://github.com/MarkusBordihn/eslint-config-google/actions/runs/15162705546

MarkusBordihn commented:
As soon as this PR is merged (since I do not have permission to merge it myself), I will update the corresponding npm package on [npmjs.com](https://www.npmjs.com/).
If any additional review or approval is required before merging, please let me know.

JamesVanWaza commented:
@iennae kindly review and update

luciemdx commented:
friendly ping @iennae can you please merge?
